--- a/src/ap/acs.c
+++ b/src/ap/acs.c
@@ -558,6 +558,16 @@ static int is_common_24ghz_chan(int chan
 #define ACS_24GHZ_PREFER_1_6_11 0.8
 #endif /* ACS_24GHZ_PREFER_1_6_11 */
 
+static int acs_radar_chan(const struct hostapd_channel_data *chan)
+{
+	const int allowed[] = { 52, 56, 60, 64 };
+	unsigned int i;
+	for (i = 0; i < ARRAY_SIZE(allowed); i++)
+	if (chan->chan == allowed[i])
+		return 1;
+	return 0;
+}
+
 /*
  * At this point it's assumed chan->interface_factor has been computed.
  * This function should be reusable regardless of interference computation
@@ -636,6 +646,9 @@ acs_find_ideal_chan(struct hostapd_iface
 		if (!chan_pri_allowed(chan))
 			continue;
 
+		if (acs_radar_chan(chan))
+			continue;
+
 		if (!is_in_chanlist(iface, chan))
 			continue;
 
